src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 500;
+    this.name = 'Test';
+    Animal.alive.push(this.name);
+  }
+  static deleteIndex = Animal.alive.findIndex((animal) => animal.health !== 'dead');
+
+  deleteAnimal() {
+    Animal.alive = Animal.alive.filter((animal) => animal.name = 'dad');
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    this.hidden = true;
+  }
+  hide() {
+    return this;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden === false && animal instanceof Animal) {
+      animal.health -= 50;
+    }
```

Comment:
The bite method should check if the animal is an instance of Herbivore before attempting to bite. Also, it should not decrease health if the target animal is a Carnivore or if the herbivore is hidden.